src/factories/routePathFactory.ts

```diff
@@ -19,26 +25,34 @@ class RoutePathFactory {
             ].join('-'),
         ).toString();
 
-        const nodes:INode[]
+        const routePathLinks:IRoutePathLinkResult[]
         = suunta.reitinlinkkisByReitunnusAndSuuvoimastAndSuusuunta.edges
-            .map((node: any) => NodeFactory.createNode(internalRoutePathId, node.node));
+            .map((node: any) =>
+                RoutePathLinkFactory.createRoutePathLink(node.node));
 
         const coordinates = JSON.parse(suunta.geojson).coordinates;
         const positions = coordinates.map((coor: [number, number]) => [coor[1], coor[0]]);
 
-        return {
+        const routePath : IRoutePath = {
             routeId,
-            nodes,
             positions,
-            geoJson: JSON.parse(suunta.geojson),
+            routePathLinks: routePathLinks.map(res => res.link),
```

Comment:
One of these should probably be named different. Now they seem like the same thing when they actually aren't.